gsms.py

```diff
@@ -153,6 +157,176 @@ def get_win_path(folder_id: str) -> str:
     # Free memory used by pointer
     _CoTaskMemFree(path_pointer)
 
+    print(path)
+
+    return path
+
+
+def copy_image(src_image: str, dst_image: str) -> None:
+    """
+    Copies an image from src_image to dst_image if the dst is empty or different
+
+    Parameters
+    ----------
+    src_image: str
+        Source path of the image
+    dst_image: str
+        Destination path of the image
+
+    Returns
+    -------
+    None
+
+    Examples
+    --------
+    >>> copy_image("C:\\Image1.png", "D:\\Image1.png")
+    """
+    # if src_image exists and dst_image does not exist
+    if os.path.isfile(src_image) and not os.path.isfile(dst_image):
+        shutil.copy2(src=src_image, dst=dst_image)  # copy2 preserves metadata
+        print(f'Copied box-art image to: {dst_image}')
+    else:
+        print(f'No box-art image found at: {src_image}')
+
+
+def has_app(sunshine_apps: any, name: str) -> bool:
+    """
+    Checks if a given app name is in the sunshine_apps object
+
+    Parameters
+    ----------
+    sunshine_apps: any
+        Parsed JSON object of the sunshine `apps.json`
+    name: str
+        Name to check for
+
+    Returns
+    -------
+    bool:
+        True if the app is in the sunshine_apps object. Otherwise False
+
+    Examples
+    --------
+    >>> has_app(any, "Game Name")
+    False
+    """
+    app_exists = False
+
+    for existing_app in sunshine_apps['apps']:
+        if name == existing_app['name']:
+            app_exists = True
+            print(f'{name} app already exist in Sunshine apps.json, skipping.')
+            break
+
+    return app_exists
+
+
+def add_game(sunshine_apps: any, name: str, logfile: str, cmd: str, working_dir: str, image_path: str) -> None:
+    """
+    Function to add an app to the sunshine_apps object passed in
+
+    Parameters
+    ----------
+    sunshine_apps: any
+        Parsed JSON object of the sunshine `apps.json`
+    name: str
+        Name of the app
+    logfile: str
+        Logfile name for the app
+    cmd: str
+        Commandline to start the app
+    working_dir: str
+        Working directory for the app
+    image_path: str
+        Path to an image file to display as box-art
+
+    Returns
+    -------
+    None
+
+    Examples
+    --------
+    >>> add_game(any, "Game Name", "game.log", "game.exe", "C:\\gamedir", "C:\\gamedir\\image.png")
+    """
+
```

Comment:
I haven't tested this, but what about a loop like this? Instead of repeating the code?

```py
path_checks = [ "cmd", "working_dir" ]

for _ in path_checks:
    locals()[_] = known_path_to_absolute(locals()[_])

    # remove preceding separator on the path if it exists
    while locals()[_].endswith(os.sep):
        locals()[_] = locals()[_][:-1]
```

Reply:
this would only be replacing the 2 `known_path_to_absolute` function calls as the 2 loops to remove separators work on different ends depending on the variable. for the working dir we check the end for the command we check the start.